### PR TITLE
Fixes for #106

### LINKED ISSUE #106: Deprecation warnings to streamline migration from 0.3 to 0.4
Pandana 0.4 made some [changes to the API](https://github.com/UDST/pandana/releases/tag/v0.4.0) that cause older user code to crash. We're still running into this in UrbanSim models, and it's hard for people to diagnose the errors if they're not expecting them.

In the next point release, I think we should make some small adjustments to allow older code to continue running, raising deprecation warnings to explain the changes that need to be made:

### 1. `pandana.network.reserve_num_graphs()`

This was removed in 0.4 because improvements to the memory management mean that it's no longer necessary. But it appears frequently in older user code.

#### Current behavior: 
- code that calls this function crashes, usually with an `AttributeError`, because the function no longer exists

#### Proposed behavior:
- allow the function call, but ignore it and raise a `DeprecationWarning` alerting users that it's no longer necessary

### 2. `pandana.Network().init_pois()` and `set_pois()`

In 0.4 `init_pois()` was removed, and two of its arguments were shifted to `set_pois()`. 

#### Current behavior:
- code that calls `init_pois()` crashes
- code that calls `set_pois()` expecting the older API crashes with a `TypeError` because not enough arguments are provided

#### Proposed behavior:
- one of the shifted arguments is currently [saved as a class attribute](https://github.com/UDST/pandana/blob/v0.4.1/pandana/network.py#L485-L493), and i don't see a problem with saving the other one as well
- i think we should allow calls to `init_pois()`, save the two relevant arguments as class attributes, and raise a `DeprecationWarning` alerting users of the API change
- `set_pois()` should read those two arguments from the class attributes if they're not provided, and raise a `DeprecationWarning` explaining that they shouldn't be provided through `init_pois()` any more

### Discussion

Making these changes would allow code written for Pandana 0.3 to continue running, while explaining to users how to make the appropriate updates. I think this would be pretty valuable and save some headaches.

For example, it looks like `urbansim` itself still [pins](https://github.com/UDST/urbansim/blob/v3.1.1/setup.py#L52) its `pandana` requirement at 0.3, presumably because of the API changes -- this should make migration easier.

### PR DESCRIPTION
**Pull request by alephcero**
Fixes #106 https://github.com/UDST/pandana/issues/106

### CODE REVIEW

File: pandana/network.py [hunk 1]
```diff
@@ -455,7 +458,34 @@ def plot(
...
+            units such as utility or time etc.
+        max_pois :
+            Maximum number of POIs to return in the nearest query
+
+        """
+        self.num_categories = num_categories
+        self.max_dist = max_dist
+        self.max_pois = max_pois
+        warnings.warn(
+            "init_pois will be deprecated, num_categories and max_dist \
```
Comment (smmaurer) [inline pandana/network.py]: Let's make this a `DeprecationWarning` with the message "Method init_pois() is no longer needed in Pandana 0.4+ and will be removed in a future version; maxdist and maxitems should now be passed to set_pois()"

**Comment by smmaurer** [comment]
As i'm reviewing, i realized there's one code migration case that's kind of tricky to handle. @sablanchard @alephcero 

In Pandana 0.3, we had:
`set_pois(str category, pd.Series x_col, pd.Series y_col)`

In Pandana 0.4, we have:
`set_pois(str category, int maxdist, int maxitems, pd.Series x_col, pd.Series y_col)`

Apologies for my ad-hoc type annotations, but you get the idea. As of this PR, `maxdist` and `maxitems`  are taken automatically from `init_pois()` to allow old code to continue running in both versions. But if the old code is passing values by _position_ without including the parameter name, Pandana 0.4 will still crash at `set_pois()` when it tries to use a series as an integer. Two options:

1. We can check if `maxdist` is a `pd.Series` and if so, use that value for `x_col` instead. And same for `maxitems` and `y_col`. All parameters are required, so we don't need to worry about anything being missing. 

2. Or we can check if `maxdist` is a `pd.Series` and if so, raise an error telling people to update their code. The update would be easy and cross-compatible because all they need to do is pass the arguments by name.

I guess i'd lean toward option 1 because it's so seamless for users. It seems kind of risky to switch parameters around like that, but i can't think of any actual problems it could cause! What do you guys think?

**Comment by federicofernandez** [comment]
> 1. We can check if `maxdist` is a `pd.Series` and if so, use that value for `x_col` instead. And same for `maxitems` and `y_col`. All parameters are required, so we don't need to worry about anything being missing.
> 2. Or we can check if `maxdist` is a `pd.Series` and if so, raise an error telling people to update their code. The update would be easy and cross-compatible because all they need to do is pass the arguments by name.
> 
> I guess i'd lean toward option 1 because it's so seamless for users. It seems kind of risky to switch parameters around like that, but i can't think of any actual problems it could cause! What do you guys think?

What about doing (1) but also showing a deprecation warning?

**Comment by smmaurer** [comment]
> What about doing (1) but also showing a deprecation warning?

We'll raise a deprecation warning when `init_pois` runs, so it will show up either way.

`DeprecationWarning: Method init_pois() is no longer needed in Pandana 0.4+ and will be removed in a future version; maxdist and maxitems should now be passed to set_pois()`

**Comment by federicofernandez** [comment]
> > What about doing (1) but also showing a deprecation warning?
> 
> We'll raise a deprecation warning when `init_pois` runs, so it will show up either way.
> 
> `DeprecationWarning: Method init_pois() is no longer needed in Pandana 0.4+ and will be removed in a future version; maxdist and maxitems should now be passed to set_pois()`

I was referring to explicitly say to the user that we are re-interpreting the parameters, that is the scary part from my point of view. I know that it'll work but if then a user looks the documentation and the code might get confused. But of course, we can also very explicit about this in the documentation too.

File: pandana/network.py [hunk 2]
```diff
@@ -90,6 +86,13 @@ def __init__(self, node_x, node_y, edge_from, edge_to, edge_weights,
 
         self.kdtree = KDTree(nodes_df.as_matrix())
 
+    def reserve_num_graphs(self, num):
+        warnings.warn(
+            "reserve_num_graphs is deprecated, as pandana 4.1. There is no need for it any more",
```
[the remark this answers is not in the document]
Comment (alephcero) [reply pandana/network.py]: Done! I had to use `warnings.simplefilter("always")` in my script for the warning message to be shown.

**Pull request merged**